addons/mail/static/src/composer/composer.js

```diff
@@ -172,6 +180,17 @@ export class Composer extends Component {
         });
     }
 
+    get isDisabled() {
+        if (!this.thread || this.thread.type !== "chatter") {
+            return false;
+        }
+        if (this.thread.hasWriteAccess === undefined) {
+            this.threadService.fetchAccessRights(this.thread);
+            return false;
+        }
+        return this.thread.isReadOnly;
+    }
```

Comment:
It's too late to check access rights when showing the composer. We should show the composer when the user can compose a message. If the user cannot post, then there's no reason to show the composer.
- In chatter, "Send message"/"Log note" buttons should be disabled, preventing opening the composer.
- message actions should not show `Reply` action if the user cannot actually reply to the message

Ok for Discuss app & Chat window to show disabled composer, though the footprint could ideally be way smaller, with just a one-line small text that tells to user cannot post.